app/models/reply.rb

```diff
@@ -118,4 +118,15 @@ def update_post_authors
     return if is_import
     post.user_joined(user)
   end
+
+  def remove_post_author
+    return if post.user == user
+    return unless post.replies.where(user: user).blank?
```

Comment:
`return if post.replies.where(user: user).exists?` doesn't load the objects, so would be preferable.